### PR TITLE
Add service metadata from stack spec to service generated by stack

### PR DESCRIPTION
**Pull request by chandu188**
Fixes https://github.com/zalando-incubator/stackset-controller/issues/222

### CODE REVIEW

**Comment by chandu188** [comment]
@szuecs Could you give me some feedback on the PR?

File: pkg/core/stack_resources.go [hunk 1]
```diff
@@ -248,8 +248,14 @@ func (sc *StackContainer) GenerateService() (*v1.Service, error) {
 		return nil, err
 	}
 
+	metaObj := sc.resourceMeta()
+	stackSpec := sc.Stack.Spec
+	if stackSpec.Service != nil {
+		metaObj.Annotations = mergeLabels(metaObj.Annotations, stackSpec.Service.Annotations)
+		metaObj.Labels = mergeLabels(metaObj.Labels, stackSpec.Service.Labels)
```
Comment (mikkeloscar) [inline pkg/core/stack_resources.go:255]: On ingress we only insert annotations, maybe we should do the same here. I don't see a need for inserting labels so let's keep it out until we have a use case for it IMO.

Comment (cjbooms) [reply pkg/core/stack_resources.go:255]: You sure about this @mikkeloscar I see labels being propagated onto StackSet owned `Ingress` resources. EG:
```
kind: Ingress
metadata:
  annotations:
...
  labels:
    application: rules-evaluator
    deployment-id: d-uossw3sit9rfgwhj776zv84ky
    pipeline-id: l-28ykzm3xih43dne4518ubhjs3i
    stackset: rules-evaluator
```

Comment (cjbooms) [reply pkg/core/stack_resources.go:255]: Oh nevermind, I see the difference. You're referring to labels defined on the Service itself

**Comment by chandu188** [comment]
@mikkeloscar removed labels as per the suggestion !!

**Comment by mikkeloscar** [comment]
:+1:

**Comment by cjbooms** [comment]
Any ETA for when this will be merged? I'm hoping to benchmark a new API for Pop and would like my Service annotation to get picked up so I can differentiate the metrics:
```
  stackTemplate:
    spec:
      service:
        metadata:
          annotations:
            zalando.org/api-usage-monitoring-tag: "load-test"
```

**Comment by szuecs** [comment]
:+1:

**Pull request merged**